### PR TITLE
global: fix tests from pallets family upgrade

### LINKED ISSUE #273: ORCiD and Keycloak integrations not working
Reported by @dfdan and confirmed it happens in CERN login (Keycloak) too.

```
File "/srv/venv/kittler-lectures/lib/python3.8/site-packages/invenio_accounts/models.py", line 138, in __init__
    super().__init__(*args, **kwargs)
  File "/srv/venv/kittler-lectures/lib/python3.8/site-packages/sqlalchemy/orm/decl_base.py", line 1163, in _declarative_constructor
    raise TypeError(
TypeError: 'profile' is an invalid keyword argument for User
```

### PR DESCRIPTION
**Pull request by ppanero**
- closes #273 

https://user-images.githubusercontent.com/6756943/169026942-3fb8721e-41af-4869-9b77-d6037e2d8c02.mov

# Important info

All redirects are now relative, since Flask/Werkzeug 2.1.0. According to the [issue](https://github.com/pallets/werkzeug/issues/2352#issuecomment-1070977357) it was tested in all the major browsers and CLIs and they do follow redirects properly.

> Response.autocorrect_location_header is disabled by default. The Location header URL will remain relative, and exclude the scheme and domain, by default. `wekzeug/issues/2352`

In order to undo it (i.e. allow external redirects) we would need to change [this configuration value](https://github.com/pallets/flask/blob/81c7ff7f8db09b5d609a09722785f70ec6f7af68/src/flask/wrappers.py). An example test in werkzeug is [this](https://github.com/pallets/werkzeug/blob/76de1b82c7ef9c692cf3731a4de53d7b213f3bf7/tests/test_test.py#L470)

### CODE REVIEW

File: invenio_oauthclient/contrib/cern_openid.py [hunk 1]
```diff
@@ -363,7 +362,6 @@ def account_setup(remote, token, resp):
     resource = get_resource(remote, resp)
 
     with db.session.begin_nested():
-        person_id = resource.get("cern_person_id")
```
Comment (ppanero) [inline invenio_oauthclient/contrib/cern_openid.py:366]: unused

Comment (kpsherva) [reply invenio_oauthclient/contrib/cern_openid.py:366]: this is interesting, we have person_id in ILS, not sure now how it gets there

File: invenio_oauthclient/contrib/globus.py [hunk 2]
```diff
@@ -230,7 +230,8 @@ def account_info(remote, resp):
             'email': info['email'],
             'profile': {
                 'username': info['username'],
-                'full_name': info['name']
+                'full_name': info['name'],
+                'affiliation': ''
```
Comment (kpsherva) [inline invenio_oauthclient/contrib/globus.py:234]: is it still needed to hardcode to empty string ?

File: invenio_oauthclient/utils.py [hunk 3]
```diff
@@ -163,15 +176,21 @@ def oauth_register(form, user_info=None, precedence_mask=None):
...
             default_mask = {"email": True}
+            # act on form data so the `profile` is updated correctly
             filter_user_info(user_info, precedence_mask or default_mask)
-            patch_dictionary(data, user_info)
+            patch_dictionary(form_data, user_info)
 
+        # re-populate the form after appyling the precedence mask and
+        # convert the form data to user model's data
+        data = fill_form(form, form_data).to_dict()
+        # TODO: check why None values are there in the first place
```
Comment (kpsherva) [inline invenio_oauthclient/utils.py:192]: yeah, could we maybe create a ticket ?

File: invenio_oauthclient/contrib/globus.py (hunk 2 above)
Comment (ppanero) [reply invenio_oauthclient/contrib/globus.py:234]: yes, nice catch!

File: invenio_oauthclient/utils.py (hunk 3 above)
Comment (ppanero) [reply invenio_oauthclient/utils.py:192]: https://github.com/inveniosoftware/invenio-oauthclient/issues/275

File: invenio_oauthclient/contrib/cern_openid.py (hunk 1 above)
Comment (ppanero) [reply invenio_oauthclient/contrib/cern_openid.py:366]: 🤷‍♂️ I just see that the `person_id` is extracted and never used hehe

File: invenio_oauthclient/contrib/keycloak/handlers.py [hunk 4]
```diff
@@ -62,6 +62,7 @@ def info_handler(remote, resp):
             "profile": {
                 "full_name": user_info.get("name"),
                 "username": user_info.get("preferred_username"),
+                "affiliation": '',
```
Comment (kpsherva) [inline invenio_oauthclient/contrib/keycloak/handlers.py:65]: here as well

**Review by kpsherva — APPROVED** [review]

**Pull request merged**